03week/ticTacToe.js

```diff
@@ -14,36 +18,155 @@ let board = [
+}
+
+const horizontalWin = () => {
+
+  for(let i = 0;i<board.length;i++){
+    return (isArrayAllEqual(board[i]))
+    // go through every row and take that array and send it to our array tester
+    // easiest case since they are all in the same array
+    // console.log("Horizontal Win")
+    // return 1;
+  }
+}
 
-function horizontalWin() {
-  // Your code here
+const verticalWin = () => {
+  for(let i =0;i<board.length;i++){
+    let stringOfVerticleElements = "";
+    // making a string to turn into an array cause I didn't want to push to a temp array
+    // vestigial c++ methods shining through 
+    for(let j =0;j<board.length;j++){
+      stringOfVerticleElements+=board[j][i];
+    }
+    if(stringOfVerticleElements.trim() != "" && isArrayAllEqual(stringOfVerticleElements.split(""))){
+      // funny big of logic, but prevents trying to make an array out of empty string.
+      // I end up double testing for empty string a lot of places, need to clean that up
+      // console.log("Vertial Win")
+      // keeping around for testing, delete when I can
+      return 1;
+    }
+    stringOfVerticleElements = ""
+    // reset the string builder
+    // I don't like the way I sloved this, likely to refactor
+  }
 }
 
-function verticalWin() {
-  // Your code here
+const diagonalWin = () => {
+  //left to right diaginal
+  let stringToArrayHolder = "";
+  //another string to array builder
+  for(let i =0;i<board.length;i++){
+    stringToArrayHolder+=board[i][i]
+  }
+  if(stringToArrayHolder.trim() != "" && isArrayAllEqual(stringToArrayHolder.split(""))){
+    console.log("Diginal Win left to right")
+    return 1;
+  }
+  stringToArrayHolder="";
+  //right to left diaginal
+  for(let j =0;j<board.length;j++){
+    stringToArrayHolder+=board[j][(board.length-1)-j]
+  }
+  if(stringToArrayHolder.trim() != "" && isArrayAllEqual(stringToArrayHolder.split(""))){
+    console.log("Diginal Win right to left")
+    return 1;
+  }
 }
 
-function diagonalWin() {
-  // Your code here
+const checkForWin = () => {
+
+  return verticalWin() || horizontalWin() || diagonalWin();
 }
 
-function checkForWin() {
-  // Your code here
+
+const whatIsAtPosition = (row, column)=>{
+  if (row >= 0 && row < 3 && column>=0 && column<3){
+    return board[row][column];
+  }
+  return "err"
+
+  // This is tricky, there are 3 values this returns that are needed, an X/O, a blank or an 
+  // index out of bounds. To prevent things from blowing up, I had to use
+  // some short circut logic
 }
 
-function ticTacToe(row, column) {
-  // Your code here
+const freeSpots= () =>{
+  return board[0].filter(boardSpot => boardSpot == " ").length + 
+  board[1].filter(boardSpot => boardSpot == " ").length + 
+  board[2].filter(boardSpot => boardSpot == " ").length;
 }
 
-function getPrompt() {
+const switchPlayer = (playerTurn) => {
+  if(playerTurn == "X"){
+    return "O"
+  }
+  return "X"
+}
+
+
+const ticTacToe = (row, column) => {
+  const formatedRow = parseInt(row);
```

Comment:
intParse is a great to control converting the number however you need to control whether i can input a decimal, letter, or negative